src/Login.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import cookie from 'react-cookie';
+
+import {
+  Button,
+  ControlLabel,
+  Form,
+  FormControl,
+  FormGroup,
+} from 'react-bootstrap';
+import { browserHistory } from 'react-router';
+import axios from 'axios';
+
+const api = (process.env.REACT_APP_API);
+
+const validateInput = (input) => {
+  if (!(input.login)) {
+    return false;
+  }
+  if (!(input.password)) {
+    return false;
+  }
+  return true;
+}
+
+class Login extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      login: cookie.load('login') || '',
+      password: '',
+      failedValidation: false,
+    };
+    this.onChange = this.onChange.bind(this);
+    this.onLoginChange = this.onLoginChange.bind(this);
+    this.onPasswordChange = this.onPasswordChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  onChange(key) {
+    return (value) => {
+      this.setState({
+        [key]: value,
+        failedValidation: false,
+      });
+    };
+  }
+
+  onLoginChange(ev) {
+    this.onChange('login')(ev.target.value);
+  }
+
+  onPasswordChange(ev) {
+    this.onChange('password')(ev.target.value);
+  }
+
+  onSubmit(ev) {
+    ev.preventDefault();
+    if (!validateInput(this.state)) {
+      this.setState({
+        failedValidation: true,
+      })
+      return;
+    }
+    axios.post(`${api}/users`, {
+      login: this.state.login,
+      password: this.state.password
+    })
+    .then((response) => {
+      const { login, display_name } = response.data;
+      cookie.save('login', login);
+      cookie.save('display_name', display_name);
+      cookie.save('isLoggedIn', true);
```

Comment:
I'm not concerned in the short term.  It's really about what is rendered at this point.